asllib/libdir/stdlib.asl

```diff
@@ -119,15 +119,15 @@ end;
 
 func Replicate{N,M}(x: bits(M)) => bits(N)
 begin
-  assert (N MOD M == 0);
-  if N == 1 then
+  if M == 1 then
```

Comment:
What is the purpose of this? If the purpose of the code is to serve as an example, there's no need for optimizations.

Reply:
This code is called in mode `-variant asl` and the optimisation had some impact. I do not know whether this is still the  case or not.

Reply:
My preference would be not to include this optimisation in example/reference code, but I can live with it.

Reply:
We can opt to include this optimization in an internally-used version of the standard library.